### PR TITLE
Fix android queries

### PR DESCRIPTION
**Pull request by JLHwung**
I recommend to review this PR by commits.

The first commits merge chrome >= 37 data to android under the `mobileToDesktop` flag

The second commits fixed result sorting issues occured when I wrote test for the first commit.

I propose in In the next major release we should set the defaults of `opts.mobileToDesktop` to be `true` so that the returned mobile browsers are complete. People can disable this feature if they find things are not going well.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -367,11 +399,13 @@ function browserslist (queries, opts) {
...
     name2 = name2.split(' ')
     if (name1[0] === name2[0]) {
-      if (FLOAT_RANGE.test(name1[1]) && FLOAT_RANGE.test(name2[1])) {
-        return parseFloat(name2[1]) - parseFloat(name1[1])
-      } else {
-        return compare(name2[1], name1[1])
-      }
+      // here we assume that caniuse version ranges never overlaps,
+      // so it is safe to use the left of the range
+      // eslint-disable-next-line max-len
```
Comment (ai) [inline index.js]: Nope, you can't just disable the rule. Try to reorganize the code to make it more readable.

File: test/range.test.js [hunk 2]
```diff
@@ -8,6 +8,12 @@ beforeEach(() => {
       name: 'ie',
       released: ['8', '9', '10', '11'],
       versions: ['8', '9', '10', '11']
+    },
+    android: {
+      name: 'android',
+      released: ['4.2-4.3', '4.4',
```
Comment (ai) [inline test/range.test.js]: Strange new line

File: test/range.test.js [hunk 3]
```diff
@@ -24,6 +30,13 @@ it('selects versions with query out of range', () => {
   expect(browserslist('ie 1-12')).toEqual(['ie 11', 'ie 10', 'ie 9', 'ie 8'])
 })
 
+it('selects a range of android browsers', () => {
+  expect(browserslist('android 4.3-37'))
+    .toEqual([
```
Comment (ai) [inline test/range.test.js]: You can keep `toEqual` on previous line to make method smaller

File: index.js [hunk 4]
```diff
@@ -191,14 +191,42 @@ function byName (name, context) {
...
+      var cloned = cloneData(desktopData)
+      cloned.name = name
+      return cloned
+    }
   }
   return browserslist.data[name]
 }
 
+function normalizeAndroidVersions (androidVersions, chromeVersions) {
+  var firstEvergreen = 37
```
Comment (ai) [inline index.js]: If we use this bar in multiple places, we need to move it to `var ANDROID_FIRST_EVERGREEN` constant.

File: test/range.test.js (hunk 3 above)
Comment (JLHwung) [reply test/range.test.js]: 😛I am really not good at formatting without the help of prettier. If you are okay with prettier, I would suggest we leave it as-is. I can introduce prettier in a separate PR.

Comment (ai) [reply test/range.test.js]: Don't worry, fix the logic and `// eslint-disable` and I will fix the formatting.

File: test/range.test.js (hunk 2 above)
Comment (JLHwung) [reply test/range.test.js]: Good catch, addressed in bd31b51.

**Comment by ai** [comment]
Thanks. Released in 4.8.1.

**Pull request merged**